client/components/ReviewsRender.jsx

```diff
@@ -1,11 +1,65 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+const wordConditionFinder = (text, filterCondition) => {
+  const results = [];
+  const findAllInstances = (word) => {
+    const lowerCaseWord = word.toLowerCase();
+    const lowerFilterCondition = filterCondition.toLowerCase();
+    if (lowerCaseWord.indexOf(lowerFilterCondition) === -1) {
+      return results;
+    }
+    results.push(
+      word.substring(lowerCaseWord.indexOf(lowerFilterCondition),
```

Comment:
m number of invocations of the lowerCaseWord.indexOf(lowerFilterCondition) means O(m*n) where n === lowerCaseWord.length. Consider storing the resulting index in a variable if possible.